homeassistant/components/supla/switch.py

```diff
@@ -34,5 +34,5 @@ def is_on(self):
         """Return true if switch is on."""
         state = self.channel_data.get("state")
         if state:
-            return state["on"]
+            return state.get("on")
```

Comment:
Why does this need to be changed?

Reply:
To avoid exception in case "on" property is not available

Reply:
When would that be the case?

Reply:
Changes in the API or something else unexpected by me. Do you think exception in such case would be better? Is there any other significant reason to use brackets?

Reply:
No, I think it is fine. I was just curious on why you changed it.

Reply:
I'm siding on with @springstan on this one -- I think it's better to fail when the API we use changes. A partially working system is IMHO worse, than the one failing in a clear way.  So I would leave the brackets alone :) Other than that, whole PR seems solid and in line with what was done before. Good job!

Reply:
I was thinking of distant a future with nobody to maintain this anymore. Some regular users could be affected by then. No need to future-proof it that much though, so I reverted the change.